aria/modeling/service_template.py

```diff
@@ -1864,13 +1865,21 @@ def as_raw(self):
 
     def instantiate(self, container):
         from . import models
-        plugin = self.plugin_specification.find_plugin() \
-            if self.plugin_specification is not None else None
+        if self.plugin_specification and self.plugin_specification.enabled:
+            plugin = self.plugin_specification.plugin
+            implementation = self.implementation if plugin is not None else None
+            # "plugin" would be none if a match was not found. In that case, a validation error
```

Comment:
im not sure i understood this comment.
i think you're trying to say that `plugin` can't be None here since the test for enabled has already been performed earlier?
the first line of the comment makes it sound like `plugin` can be None when it cant